src/util/customErrors.ts

```diff
@@ -0,0 +1,13 @@
+/**
+ * A custom error that holds additional information and a more explicit message
+ * to simplify tracking down errors that occur during execution
+ */
+export class AnnotatedError extends Error {
+  constructor(message: string, expressionName: string, libraryName: string, localId?: string) {
+    super(
+      `Expression ${expressionName} from CQL Library: ${libraryName} ${
```

Comment:
I wouldn't have given it much thought myself, but now that you got me thinking... Yeah, a little more structured output might be nice. I went even a step further:

```
AnnotatedError: Encountered unexpected error during execution.

	Error Message:	You done messed up.
	CQL Library:	AgeAtMP|1
	Expression:	CalculateAgeAt
	ELM Local Id:	12345

    at CalculateAgeAt.execute (/Users/cmoesel/git/cqframework/cql-execution/lib/elm/expression.js:40:19)
    at /Users/cmoesel/git/cqframework/cql-execution/lib/elm/expression.js:48:45
```

You don't have to use that, but if you want to, here's the code:
```ts
super(
  `Encountered unexpected error during execution.\n\n\tError Message:\t${message}\n\tCQL Library:\t${libraryName}\n\tExpression:\t${expressionName}${
    localId ? `\n\tELM Local Id:\t${localId}` : ``
  }\n`
);
```

Reply:
Pushed this change, but I forgot to update the unit tests. Obviously I could just update the expected messages to have the new format. Is it worth making this more maintainable though? Maybe using a regex in the assertion instead to capture the specific pieces of information that we expect to be in the error, rather than a strict assertion on the entire string itself?

Reply:
Potential alternative approach to testing:

```ts
try {
  testExpression.execute(this.ctx);
  fail();
} catch (e: any) {
  should(e).be.instanceOf(AnnotatedError);
  should(e.message).containEql('Error Message:\tExecution Failed!');
  should(e.message).containEql('CQL Library:\tTestSnippet|1');
  should(e.message).containEql('Expression:\tFailingExpression');
  should(e.message).containEql('ELM Local ID:\ttest');
}
```

Reply:
> Is it worth adding a message if the localId is undefined suggesting enabling localIds in ELM translation to better locate the error? 

I don't have a strong opinion on this. It partially depends on how likely it is that the user who sees the error has control over the CQL-to-ELM translation process... Personally, I think I'd just leave it as it is -- and if people contact us for help and the expression use is ambiguous, we could ask them to recompile w/ local id.

Reply:
> Maybe using a regex in the assertion instead to capture the specific pieces of information that we expect to be in the error, rather than a strict assertion on the entire string itself?

I think that's a fine approach -- and one that we use in my other projects. Sometimes, if I want to be sure the full message structure is correct, then I'll use the full message in _one_ unit test but the targeted regexes for all the others. That way if you intentionally tweak the message, there is only one unit test to fix; but if you _unintentionally_ tweak it, you still get a failed test to let you know.